dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/utils/DependentExecute.java

```diff
@@ -267,12 +300,54 @@ public DependResult getModelDependResult(Date currentTime, int testFlag) {
             DependResult dependResult = getDependResultForItem(dependentItem, currentTime, testFlag);
             if (dependResult != DependResult.WAITING && dependResult != DependResult.FAILED) {
                 dependResultMap.put(dependentItem.getKey(), dependResult);
+                mergeVarPool(dependentItem);
             }
+            dependItemVarPoolPropertyMap.clear();
+            dependItemVarPoolEndTimeMap.clear();
             dependResultList.add(dependResult);
         }
         return DependentUtils.getDependResultForRelation(this.relation, dependResultList);
     }
 
+    /**
+     * add varPool of dependentItem to dependTaskVarPoolMap
+     *
+     * @param dependentItem
+     * @return
+     */
+    private void mergeVarPool(DependentItem dependentItem) {
+        if (dependentItem.getParameterPassing() && !dependItemVarPoolPropertyMap.isEmpty()) {
+            addTaskVarPool(dependItemVarPoolPropertyMap, dependItemVarPoolEndTimeMap, dependTaskVarPoolPropertyMap,
+                    dependTaskVarPoolEndTimeMap);
+        }
+    }
+
+    /**
+     * add varPool from dependItemVarPoolMap to dependTaskVarPoolMap
+     *
+     * @param dependItemVarPoolPropertyMap dependItemVarPoolPropertyMap
+     * @param dependItemVarPoolEndTimeMap dependItemVarPoolEndTimeMap
+     * @param dependTaskVarPoolPropertyMap dependTaskVarPoolPropertyMap
+     * @param dependTaskVarPoolEndTimeMap dependTaskVarPoolEndTimeMap
+     */
+    public static void addTaskVarPool(Map<String, Property> dependItemVarPoolPropertyMap,
```

Comment:
Can we avoid static method definition here? BTW, that would be best if you can find some way to simplify this merge logic.

Reply:
Maybe it's better to put this method in `DependentUtils.java` class ?

Reply:
Leave a comment above. In addition, if we remove static, we don't have to declare this parameters, they are all variables of this class.